lib/axiom/fs/nested_stdio.js

```diff
@@ -15,22 +15,21 @@
 import Stdio from 'axiom/fs/stdio';
 
 import ReadableStreamForwarder from 'axiom/fs/stream/readable_stream_forwarder';
+import WritableStreamForwarder from 'axiom/fs/stream/writable_stream_forwarder';
 
 /**
  * @constructor @extends {Stdio}
  * @param {!Stdio} parentStdio
  */
 export var NestedStdio = function(parentStdio) {
-  // Note: we wrap stdin, ttyin and signal so that any "onData" event handler
-  // added is scoped to the nested stdio lifetime.
+  // Note: we wrap the streams so that "onData", "onClose" and any other event
+  // handlers added are scoped to the nested stdio lifetime.
   var stdin = new ReadableStreamForwarder(parentStdio.stdin);
   var signal = new ReadableStreamForwarder(parentStdio.signal);
   var ttyin = new ReadableStreamForwarder(parentStdio.ttyin);
-  // Note: stdout, stderr and ttyout don't need to be wrapped as they don't
-  // expose events.
-  var stdout = parentStdio.stdout;
-  var stderr = parentStdio.stderr;
-  var ttyout = parentStdio.ttyout;
+  var stdout = new WritableStreamForwarder(parentStdio.stdout);
```

Comment:
Output streams now also have an event to fire (onClose), hence the need to wrap them as well.